pkg/deploy/registry/deployconfig/strategy.go

```diff
@@ -0,0 +1,81 @@
+package deployconfig
+
+import (
+	"fmt"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/runtime"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/fielderrors"
+
+	"github.com/openshift/origin/pkg/deploy/api"
+	"github.com/openshift/origin/pkg/deploy/api/validation"
+)
+
+// strategy implements behavior for DeploymentConfig objects
+type strategy struct {
+	runtime.ObjectTyper
+	kapi.NameGenerator
+}
+
+// Strategy is the default logic that applies when creating and updating DeploymentConfig objects.
+var Strategy = strategy{kapi.Scheme, kapi.SimpleNameGenerator}
+
+func (strategy) NamespaceScoped() bool {
+	return true
+}
+
+// AllowCreateOnUpdate is false for DeploymentConfig objects.
+func (strategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+// PrepareForCreate clears fields that are not allowed to be set by end users on creation.
+func (strategy) PrepareForCreate(obj runtime.Object) {
+	_ = obj.(*api.DeploymentConfig)
```

Comment:
We need a follow up issue to determine whether PrepareForCreate/Update should prevent status.latestVersion from being set out of order.  There are advantages and disadvantages.  Maybe a todo.